rpcserver.go

```diff
@@ -2181,19 +2181,52 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 
 		graph := r.server.chanDB.ChannelGraph()
 
-		// TODO: restrict number of route hints?
 		var routeHints [][]routing.HopHint
 		for _, channel := range openChannels {
+			// We'll restrict the number of individual route hints
+			// to 20 to avoid creating overly large invoices.
+			if len(routeHints) > 20 {
+				break
+			}
+
 			// Since we're only interested in our private channels,
 			// we'll skip public ones.
 			isPublic := channel.ChannelFlags&lnwire.FFAnnounceChannel != 0
 			if isPublic {
 				continue
 			}
 
-			chanID := channel.ShortChanID.ToUint64()
+			// Make sure the channel has enough capacity for our
+			// amount.
+			if amt > channel.Capacity {
```

Comment:
As the capacity might not be available for incoming payments, we should instead get the balance available to the remote peer. This doesn't have to be exact, but will at least avoid adding channels having no balance for incoming payments to the routing hints.

Reply:
Good point! Fixed.